documents/document_store.go

```diff
@@ -1,20 +1,60 @@
 package documents
 
-import "errors"
+import (
+	"errors"
+	ravenHttp "../http"
+	"github.com/google/uuid"
+)
+
+type IDocumentStore interface{
+	OpenSession() (*DocumentSession, error)
+}
 
 type DocumentStore struct{
 	Urls []string
-	DefaultDBName string
-	initialized bool
+	Database string
+	initialized, closed bool
+
+	OperationsExecutor, AdminOperationsExecutor, ServerOperationsExecutor IOperationExecutor
```

Comment:
These are exposed as Operations, Admin and Server properties.
https://github.com/ravendb/ravendb/blob/v4.0/src/Raven.Client/Documents/IDocumentStore.cs#L168